### PR TITLE
react: server config

### PR DESCRIPTION
**Pull request by aeng**
This is a change of migrating admin server configuration into reactJs.

https://zanata.atlassian.net/browse/ZNTA-2646

## Checklist

- JIRA link
- Check target branch
- Make sure all commit statuses are green or otherwise documented reasons to ignore
- QA needs to evaluate against the JIRA ticket
- Changed files and commits make sense for this PR

See [Zanata Development Guidelines](https://github.com/zanata/zanata-platform/wiki/Development-Guidelines) more for information.

----
*This template can be updated in .github/PULL_REQUEST_TEMPLATE.md*

### CODE REVIEW

**Comment by huangp** [comment]
+Reviewed +(On QA) 

---

Reviewed 21 of 21 files at r1.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/927#-:-LG3seZnAnLiNcA8IGR4:b-8op9eg)*

**Comment by huangp** [comment]
-(Awaiting review) +Reviewed +(On QA) 

---

Reviewed 4 of 4 files at r2.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/927#-:-LG83SidAGawAV8ZJNfS:b-1amvkm)*

**Comment by seanf** [comment]
Review status: all files reviewed, 2 unresolved discussions (waiting on @aeng)

---

*[server/zanata-frontend/src/tslint.json, line 23 at r2](https://reviewable.io/reviews/zanata/zanata-platform/927#-LG84aojAbwv5cT3aYls:-LG84aojAbwv5cT3aYlt:b-qssfss) ([raw file](https://github.com/zanata/zanata-platform/blob/833e0d3a9d6de8870f4d1811a0c64eefc116f5d1/server/zanata-frontend/src/tslint.json#L23)):*
> ```JSON
>         "interface-name": [true, "never-prefix"],
>         "variable-name": [true, "ban-keywords", "check-format", "allow-leading-underscore", "allow-pascal-case"],
>         "no-console": [true, "log", "info"]
> ```

Why ban just these two `console` methods? It doesn't look like you added any calls to the console anyway.

---

*[server/zanata-frontend/src/app/actions/redux-api-middleware.d.ts, line 0 at r2](https://reviewable.io/reviews/zanata/zanata-platform/927#-LG83X2rAWcP2ONrZv7T:-LG83X2rAWcP2ONrZv7U:b-6i7tfa) ([raw file](https://github.com/zanata/zanata-platform/blob/833e0d3a9d6de8870f4d1811a0c64eefc116f5d1/server/zanata-frontend/src/app/actions/redux-api-middleware.d.ts#L0)):*
Don't we already have this file at server/zanata-frontend/src/app/redux-api-middleware.d.ts?

---


*Comments from [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/927)*

**Comment by aeng** [comment]
Review status: all files reviewed, 2 unresolved discussions (waiting on @aeng)

---

*[server/zanata-frontend/src/tslint.json, line 23 at r2](https://reviewable.io/reviews/zanata/zanata-platform/927#-LG84aojAbwv5cT3aYls:-LG85bPm4JS_CjueR0eI:bp73d77) ([raw file](https://github.com/zanata/zanata-platform/blob/833e0d3a9d6de8870f4d1811a0c64eefc116f5d1/server/zanata-frontend/src/tslint.json#L23)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Why ban just these two `console` methods? It doesn't look like you added any calls to the console anyway.
</blockquote></details>

Good catch. Removing it.

---

*[server/zanata-frontend/src/app/actions/redux-api-middleware.d.ts, line  at r2](https://reviewable.io/reviews/zanata/zanata-platform/927#-LG83X2rAWcP2ONrZv7T:-LG8AMCW-dqIWix5Fxli:bss4a2b) ([raw file](https://github.com/zanata/zanata-platform/blob/833e0d3a9d6de8870f4d1811a0c64eefc116f5d1/server/zanata-frontend/src/app/actions/redux-api-middleware.d.ts#L)):*
<details><summary><i>Previously, seanf (Sean Flanigan) wrote…</i></summary><blockquote>

Don't we already have this file at server/zanata-frontend/src/app/redux-api-middleware.d.ts?
</blockquote></details>

Removed.

---


*Comments from [Reviewable](https://reviewable.io/reviews/zanata/zanata-platform/927)*

File: server/functional-test/src/test/java/org/zanata/feature/administration/ServerSettingsTest.java [hunk 1]
```diff
@@ -83,9 +82,8 @@ public void setAdministratorEmailTest() {
                 .goToAdministration()
                 .goToServerConfigPage()
                 .inputAdminEmail("lara@example.com")
-                .save();
+                .save().goToAdministration();
```
[the remark this answers is not in the document]
Comment (aeng) [reply server/functional-test/src/test/java/org/zanata/feature/administration/ServerSettingsTest.java]: Changed intentionally to be consistent with other save behavior.

**Pull request merged**